src/util/pointer_offset_size.cpp

```diff
@@ -290,6 +296,13 @@ exprt member_offset_expr(
       bit_field_bits %= 8;
       result=plus_exprt(result, from_integer(bytes, result.type()));
     }
+    else if(c.type().id() == ID_bool)
+    {
+      ++bit_field_bits;
+      const std::size_t bytes = bit_field_bits / 8;
+      bit_field_bits %= 8;
+      result = plus_exprt(result, from_integer(bytes, result.type()));
```

Comment:
Perhaps check for bytes == 0 to avoid creating a needless stack of +0s that will needlessly obscure what we're doing if not simplified away yet?